### ISSUE TITLE
[OracleReview] 5.3.3.1 SecurityScheme

### ISSUE THREAD

**Issue by egekorkan** [w3c/wot-thing-description#1613]
- [x] "For all security schemes, any private keys, passwords, or other sensitive information directly providing access should be shared and stored out-of-band and MUST NOT be stored in the TD." break down into two sentences. First should
- [x]  @type: semantic tags and semantic types should be defined in the terminology. #1655 
- [ ] default language - consider recommending english.
- [x] support: "https" seems to be a left over.
- [ ] uriVariables: precedence rules for resolving name conflicts are harmful. why not requre that all uriVariable definitions are unique within a TD?
- [x] uriVariables: The following sentence is a bit confusing: "The individual variables DataSchema cannot be an ObjectSchema or an ArraySchema."
- [x] #1644
- [x] #1643

**Comment by egekorkan** [comment]
My comment was:
- default language: I think we should not do it to avoid bad internationalization reviews.

**Comment by egekorkan** [comment]
The point about the support field is taken care of at #1558
![image](https://user-images.githubusercontent.com/20195407/180653851-1eba3fa1-d27d-4381-a507-4f146869c673.png)

**Comment by mlagally** [comment]
> * default language: I think we should not do it to avoid bad internationalization reviews.
What is the concern here? A different language being proposed?

> * uriVariables: Although it makes sense, since TD is generic we should be able to adapt to this situation where the name is used twice with different parameters.
How?
> * cancelall: Too late at this point
I'm not sure why it is too late to add this to the working draft. Sematincs are very clear.
> * writemultiple: I agree, however, there was no proper discussion on this. I could not find the issue.
Let's have the discussion here or create a separate issue.

**Comment by egekorkan** [comment]
>  uriVariables: Although it makes sense, since TD is generic we should be able to adapt to this situation where the name is used twice with different parameters.

> How?

Well if an API does that, we should be able to describe that. No need to make a restriction on it, it is not the job of this TF

**Comment by egekorkan** [comment]
> I'm not sure why it is too late to add this to the working draft. Semantics are very clear.

We cannot keep opening the spec for new features. We already do not have any real implementation of `cancelaction` and `queryaction` other than my experimental implementation. These action operations are also added since the Profile TF asked for it and given that it is not clear if we will get a Profile spec ready by the publication of the TD, I am against extending these operations even further.

**Comment by sebastiankb** [comment]
>"For all security schemes, any private keys, passwords, or other sensitive information directly providing access should be shared and stored out-of-band and MUST NOT be stored in the TD." break down into two sentences. First should

OK for me. @mmccool if this is also ok for you, would you provide a PR for this?

> default language - consider recommending english.

We already passed the I14n review. I would not touch this anymore.

> uriVariables: precedence rules for resolving name conflicts are harmful. why not requre that all uriVariable definitions are unique within a TD?

in TD 1.0 uriVariables was defined only on interaction level. So we need to keep this also for TD 1.1. In general, the precedence rules are also used by security definitions.

> uriVariables: The following sentence is a bit confusing: "The individual variables DataSchema cannot be an ObjectSchema or an ArraySchema."

ok, I think we can be more clear here. I will provide a PR

**Comment by egekorkan** [comment]
@mlagally I have checked the points that have their own issues. For the first point, I am creating a PR. The language, cancelall and uriVariable name collision seem to have the resolution of "we will not do this".

**Comment by egekorkan** [comment]
While trying to separate the security sentence into two, I have realized that it actually makes sense. We say that credentials should be out-of-band and not shared in the TD. This means the same thing, if we have two different assertions, they would have the exact same test and mean exactly the same thing. I will reword this assertion instead

**Comment by egekorkan** [comment]
I have reworded it to:
> For all security schemes, any private keys, passwords, or other sensitive information directly providing access should be shared and stored out-of-band, thus they MUST NOT be stored in the TD.

**Comment by sebastiankb** [comment]
from today's TD call, decided to close this issue.
Text in https://w3c.github.io/wot-thing-description/#securityscheme was reworded.
